src/server/handlers/firestore/consents.ts

```diff
@@ -0,0 +1,225 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the Apache License, Version 2.0 (the 'License') and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Mojaloop Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Mojaloop Foundation
+ - Name Surname <name.surname@mojaloop.io>
+
+ * Google
+ - Steven Wijaya <stevenwjy@google.com>
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+ --------------
+ ******/
+/* istanbul ignore file */
+// TODO: BDD Testing will covered in separate ticket #1702
+
+import * as uuid from 'uuid'
+import { logger } from '~/shared/logger'
+
+import { ConsentHandler } from '~/server/plugins/internal/firestore'
+import { Consent, ConsentStatus } from '~/models/consent'
+
+import { consentRepository } from '~/repositories/consent'
+import * as validator from './consents.validator'
+import config from '~/lib/config'
+import { MissingConsentFieldsError, InvalidConsentStatusError } from '~/models/errors'
+
+async function handleNewConsent(_: StateServer, consent: Consent) {
+  // Assign a consentRequestId to the document and set the initial
+  // status. This operation will create an event that triggers the execution
+  // of the onUpdate function.
+  consentRepository.updateConsentById(consent.id, {
+    consentRequestId: uuid.v4(),
+    status: ConsentStatus.PENDING_PARTY_LOOKUP,
+  })
+}
+
+async function handlePartyLookup(server: StateServer, consent: Consent) {
+  // Check whether the consent document has all the necessary properties
+  // to perform a party lookup.
+  if (!validator.isValidPartyLookup(consent)) {
+    throw new MissingConsentFieldsError(consent)
+  }
+
+  // Fields are guaranteed to be non-null by the validator.
+  try {
+    server.app.mojaloopClient.getParties(
+      consent.party!.partyIdInfo.partyIdType,
+      consent.party!.partyIdInfo.partyIdentifier
+    )
+  } catch (error) {
```

Comment:
I see this pattern everywhere -> the thrown error is logged only and exceptions aren't handled - is it intentionally? if Yes, please add some explanation in comment to hint reader.

I think the name of function `handle` is misleading - here I see that async mojaloop API is called and somewhere else we will wait on our PUT endpoint to receive requested information -> so maybe instead `handle` use `initiate` and add in comments where I can find the code which is handling the PUT response...

Reply:
makes sense 
@jgeewax  @stevenwjy  - was wondering if a specific protocol for error handling was discussed ? 

Also @eoln - I see your point re:  `handle` and have changed accordingly, but don't understand what you mean re: 'add in comments where I can find the code which is handling the PUT response...'

Reply:
I think about the short hint, something like this ->` the corresponding put handler is implemented in file abc.ts`

Reply:
ahh makes sense

Reply:
I think that's a good idea -- it was previously just logged because we had not discussed much about proper error handling and hence the code tried to focus on the happy path

Reply:
If we haven't maybe it would be a good time to make a ticket for it and reference that in comments everywhere we are logging an error that this is to be replaced?

Reply:
Following up - will do as stated above

Reply:
https://github.com/mojaloop/pisp-demo-server/issues/33